modin/experimental/engines/omnisci_on_ray/frame/data.py

```diff
@@ -259,18 +259,44 @@ def groupby_agg(self, by, axis, agg, groupby_args, **kwargs):
         if groupby_args["level"] is not None:
             raise NotImplementedError("levels are not supported for groupby")
 
-        groupby_cols = by_frame.columns.tolist()
-        agg_cols = [col for col in self.columns if col not in by_frame.columns]
+        groupby_cols = by_frame.columns
+        if isinstance(agg, dict):
+            agg_cols = agg.keys()
+        elif len(self.columns) == 1:
+            # This branch is trying to detect grouping on Series object, in that case,
+            # `by_cols == agg_cols`, so we have to explicitly set proper `agg_cols`
+            # to avoid its filtering into an empty list that happens for DataFrame cases.
+            # HACK: This check is naive and will be also true for single-column frames,
+            # however there are no more options to distinguish between Series and DataFrame
+            # at this level of abstraction. So potentially we're having a mismatch
+            # from pandas at grouping on single-column frames.
```

Comment:
We are able to tell `DataFrame` and `Series` apart at the API level, so we should be able to pass down some flag (like we do for other calls using the argument `squeeze_self`), let's not add hacks when we have working solutions.

Reply:
found out that there is a way to not use this hacky `squeeze_self` parameter. We can resolve naming conflict in a more general way by using `drop` parameter, which indicates whether `by` data came from the same frame, if it is then aggregation columns is difference between all frame columns and the "by" one, otherwise aggregation columns are all of the frame columns.